lms/static/js/student_account/views/account_settings_fields.js

```diff
@@ -0,0 +1,302 @@
+                    required: this.options.required,
+                    selectOptions: this.options.options,
+                    message: this.helpMessage
+                }));
+                this.updateValueInField();
+                return this;
+            },
+
+            fieldValue: function () {
+                return this.$('.account-settings-field-value select').val();
+            },
+
+            updateValueInField: function () {
+                this.$('.account-settings-field-value select').val(this.modelValue() || "");
+            },
+
+            saveValue: function () {
+                var attributes = {};
+                attributes[this.options.valueAttribute] = this.fieldValue();
+                this.saveAttributes(attributes);
+
+            }
+        });
+
+        AccountSettingsFieldViews.LinkFieldView = AccountSettingsFieldViews.FieldView.extend({
+
+            templateSelector: '#field_link-tpl',
+
+            events: {
+                'click a': 'linkClicked',
+            },
+
+            initialize: function (options) {
+                this._super(options);
+                _.bindAll(this, 'render', 'linkClicked');
+            },
+
+            render: function () {
+                this.$el.html(this.template({
+                    title: this.options.title,
+                    linkTitle: this.options.linkTitle,
+                    linkHref: this.options.linkHref,
+                    message: this.helpMessage,
+                }));
+                return this;
+            },
+
+            linkClicked: function () {
+                event.preventDefault();
+            },
+        });
+
+        AccountSettingsFieldViews.PasswordFieldView = AccountSettingsFieldViews.LinkFieldView.extend({
+
+            initialize: function (options) {
+                this._super(options);
+                _.bindAll(this, 'resetPassword');
+            },
+
+            linkClicked: function (event) {
+                event.preventDefault();
+                this.resetPassword(event)
+            },
+
+            resetPassword: function (event) {
+                var data = {};
+                data[this.options.emailAttribute] = this.model.get(this.options.emailAttribute);
+
+                var view = this;
+                $.ajax({
+                    type: 'POST',
+                    url: view.options.linkHref,
+                    data: data,
+                    success: function (data, status, xhr) {
+                        view.showSuccessMessage()
+                    },
+                    error: function (xhr, status, error) {
+                        view.showErrorMessage(xhr);
+                    }
+                });
+            },
+
+            successMessage: function() {
+                return this.indicators['success'] + interpolate_text(
+                    gettext('Complete you password reset using the confirmation link emailed to {email_address}.'),
```

Comment:
Text:

We've sent a message to your email address. Click the link in the message to reset your password.